drools-parser/src/main/java/org/drools/parser/DRLVisitorImpl.java

```diff
@@ -124,7 +168,38 @@ public Object visitAttribute(DRLParser.AttributeContext ctx) {
         return super.visitAttribute(ctx);
     }
 
+    @Override
+    public Object visitLhsExists(DRLParser.LhsExistsContext ctx) {
+        ExistsDescr existsDescr = new ExistsDescr();
+        currentConstructStack.peek().addDescr(existsDescr);
+        currentConstructStack.push(existsDescr);
+        try {
+            return super.visitLhsExists(ctx);
+        } finally {
+            currentConstructStack.pop();
+        }
+    }
+
+    @Override
+    public Object visitLhsNot(DRLParser.LhsNotContext ctx) {
+        NotDescr notDescr = new NotDescr();
+        currentConstructStack.peek().addDescr(notDescr);
+        currentConstructStack.push(notDescr);
+        try {
+            return super.visitLhsNot(ctx);
+        } finally {
+            currentConstructStack.pop();
+        }
+    }
+
     public PackageDescr getPackageDescr() {
         return packageDescr;
     }
+
+    private void populateStartEnd(BaseDescr descr, ParserRuleContext ctx) {
+        descr.setStartCharacter(ctx.getStart().getStartIndex());
+        // TODO: Current DRL6Parser adds +1 for EndCharacter but it doesn't look reasonable. At the moment, I don't add. Instead, I fix unit tests.
+        //       I will revisit if this is the right approach.
+        descr.setEndCharacter(ctx.getStop().getStopIndex());
+    }
```

Comment:
As written in the comment, the current `DRL6Parser` implementation adds +1 for EndCharacter so it's a next index to the EndCharacter. It doesn't look reasonable to me. So in this `DRLVisitorImpl`, I set the exact `StopIndex`. I will revisit if I will find something wrong.